src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -1141,4 +1141,12 @@ public void AddDocumentIdsWithFilePath(ref TemporaryArray<DocumentId> temporaryA
             this.AdditionalDocumentStates.GetFirstDocumentIdWithFilePath(filePath) ??
             this.AnalyzerConfigDocumentStates.GetFirstDocumentIdWithFilePath(filePath);
     }
+
+    public int CompareTo(ProjectState? other)
+    {
+        if (other is null)
+            return 1;
+
+        return ((IComparable<ProjectId>)this.Id).CompareTo(other.Id);
```

Comment:
make an internal CompareTo that we can call directly.

Reply:
Done!